File: Python/DoublePoint/moveZeroes/moveZeroes.py
def moveZeroes(nums: list[int]):
    count: int = 0
    for index, num in enumerate(nums):
        nums[index - count] = nums[index]

        if num == 0:
            count += 1

    for index in range(count):
        nums[len(nums) - count + index - 1] = 0

File: golang/DoublePoint/moveZeroes/moveZeroes.go
package movezeroes

func moveZeroes(nums []int) {
	count := 0
	for index, num := range nums {
		nums[index-count] = nums[index]

		if num == 0 {
			count++
		}
	}

	for i := len(nums) - count; i < len(nums); i++ {
		nums[i] = 0
	}
}
